### ISSUE TITLE
Align vertical angle of voxel barrels with weapon projectile Angle

### ISSUE THREAD

**Issue by reaperrr** [OpenRA/OpenRA#9511]
Needed for proper implementation of TS artillery, for example.

**Comment by GraionDilach** [comment]
Needs to be controllable though, RA2 broke/deleted this logic.

**Comment by reaperrr** [comment]
Maybe a `AlignWithArmament:` property on `WithVoxelBarrel`, which allows you to select with which `Armament` to align it with (and default would be `null`, of course).

**Comment by penev92** [comment]
I'm all *for* configurability, but doing it just because one of the original games broke the feature is silly.
Same could then go for helicopter rotors in TS or FoW in TS/RA2. ;)

**Comment by atlimit8** [comment]
`primary` would be a better default. I also don't see why brake something for bug compatibility.

**Comment by GraionDilach** [comment]
Okay, I reiterate. RA2 voxels are __not__ made with this in mind - most of them look jarring with moving barrels. Animated voxels weren't using this neither (note that recoil was __always__ a voxel animation via playing many voxels after each other in RA2 (Grand Cannon, Prism Tank)). That and separated barrels themselves fell out of use, with only the pre-RA2 Light Tank, Grizzly, Rhino and Apocalypse having such... hell, YR Lasher Tank even had a bug of having the  aforementioned pre-RA2 Light Tank barrel _along with it's own barrel merged into the turret_!

I'm all up for fixing WW bugs and everything, but this should be definitely optional - I don't mind the default value itself though as long as the option exists.

**Comment by penev92** [comment]
Fair enough. I guess RA2 will need that regardless of what we want, unless we redo the artwork, which isn't exactly fiesable.

**Comment by atlimit8** [comment]
An empty string could be usable as meaning no armament. `primary` would still be the preferred default.
